### PR TITLE
Labels for feature

### LINKED ISSUE #463: Extend feature set and/or feature metadata
This issue tracks the addition of new fields to the current feature set specification that allow a user to add metadata to either the feature set or features. These fields are optional and are intended to provide users with the flexibility to include feature level or feature level information.

The current proposal is to only add a single string field called `description` to `FeatureSpec`

### PR DESCRIPTION
**Pull request by imjuanleonard**
1. Ensure that your code follows our code conventions ✅ 
2. [Run unit tests and ensure that they are passing](https://github.com/gojek/feast/blob/master/docs/contributing.md#running-unit-tests) ✅ 
3. If your change introduces any API changes, make sure to update the integration tests scripts [here](https://github.com/gojek/feast/tree/master/tests/e2e) ✅ 
4. Make sure documentation is updated for your PR! ✅ 
5. Make sure you have signed the CLA https://cla.developers.google.com/clas ✅ 

**What this PR does / why we need it**:
Full discussion is available on [this issue](https://github.com/gojek/feast/issues/463)

**Which issue(s) this PR fixes**:
Fixes #463 on the service side

**Does this PR introduce a user-facing change?**:

```release-note
- Add labels column to Feature/Entity
```

### CODE REVIEW

**Comment by imjuanleonard** [comment]
*WIP Checklist*
-> Add Mac development set-up guide using homebrew
-> Create the label column on feature set ✅ 
-> Create a Gateway to CRUD the label column 🛑  (Currently to implement the list, I just insert it directly to the table)
-> Create the python client ✅ 
-> Verify the column types for the feature set table 🛑 
```
Python: dict()
Java: Map<String,String>
Table: Varchar(255)
```
-> Add the test with label Columns enabled 🛑 
-> Update the documentation on how to use the lable from FeatureSetClass function 🛑

File: sdk/python/feast/feature_set.py [hunk 1]
```diff
@@ -230,6 +233,23 @@ def max_age(self, max_age):
         """
         self._max_age = max_age
 
+    @property
+    def label(self):
```
Comment (woop) [inline sdk/python/feast/feature_set.py]: Label isnt a property of a feature set, its a property of a feature.

Comment (imjuanleonard) [reply sdk/python/feast/feature_set.py]: My bad, Thanks for clarifying

Hence, this is current to do Checklist:

- Add Mac development set-up guide using homebrew
- Create the `labels` column on Field Class (This will affect the Feature and Entity)  
- Create a gateway to Create, List, and Delete to the labels Column in `Feature` class using python SDK Client
- Add the test with labels Column empty and labels Column marked in Feature class
- Update the integration test tree
- Update the documentation on how to use the labels from Feature function

Notes, Datatype for each of the layers
```
Python: dict()
Java: Map<String, String>
Table: Varchar(255): with key-value object store in JSON
```

Comment (woop) [reply sdk/python/feast/feature_set.py]: `Varchar(255): with key-value object store in JSON`

Why are you limiting it to that small of a size? I also think you should maybe consider comma or tab separation here and using a Set (in the Java model)

File: sdk/python/setup.py [hunk 2]
```diff
@@ -48,6 +48,7 @@
     "numpy",
     "google",
     "confluent_kafka",
+    "tensorflow-metadata>=0.21,<0.22",
```
Comment (woop) [inline sdk/python/setup.py]: What is this for?

Comment (imjuanleonard) [reply sdk/python/setup.py]: This is a missing dependency because we are using the TensorFlow metadata proto

Previously I keep facing this bug whenever I tried to generate my proto. We verified on a clean install on @davidheryanto environment as well

Comment (woop) [reply sdk/python/setup.py]: Ok, but how is it related to your PR? Should it be a separate PR?

Comment (imjuanleonard) [reply sdk/python/setup.py]: I think you're right, should created a separate PR
https://github.com/gojek/feast/pull/508

This is the PR @woop, this also generates the python SDK Protos

This will help to reduce the file change in this PR, so we can focus on the features

File: sdk/python/feast/feature_set.py (hunk 1 above)
Comment (imjuanleonard) [reply sdk/python/feast/feature_set.py]: Sure @woop , now I am using 'TEXT' for a result similar with Varchar(Max), thanks for the input

File: core/src/main/java/feast/core/model/Field.java [hunk 3]
```diff
@@ -214,6 +225,40 @@ public Field(EntitySpec entitySpec) {
     }
   }
 
+  public Map<String, String> getLabels() {
+    ObjectMapper mapper = new ObjectMapper();
```
Comment (zhilingc) [inline core/src/main/java/feast/core/model/Field.java]: in TypeUtils we have helper functions `convertJsonStringToMap` and `convertMapToJsonString`, you might want to use those for converting to and from json.

**Comment by imjuanleonard** [comment]
All the comments in this merge have been addressed, it is ported to #536 536
